### PR TITLE
Fix/post sync

### PR DESCRIPTION
**Pull request by atiq-ur**
= 1.0.12 =
- Fix:  Legacy post types support for WPPostSync integration

### CODE REVIEW

File: src/MigrationScript.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+class MigrationScript {
+
+	private static $instance = null;
+
+	public function __construct() {
+		$this->run();
+	}
+
+	public static function instance(): MigrationScript {
+		if (!isset(self::$instance) && !(self::$instance instanceof MigrationScript)) {
```
Comment (Rijoanul-Shanto) [inline src/MigrationScript.php:16]: `!isset(self::$instance)`

if that returns true then we don't need to check `!(self::$instance instanceof MigrationScript)` right?

File: src/MigrationScript.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	public function run() {
+		$migration = get_option( 'thrivedesk_migration_version', THRIVEDESK_VERSION );
+		if ( $migration ) {
```
Comment (Rijoanul-Shanto) [inline src/MigrationScript.php:25]: this section will be always true right?

File: src/MigrationScript.php [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+			$this->runMigration();
+		}
+	}
+
+	private function runMigration( ) {
+		$this->v1_0_12();
```
Comment (Rijoanul-Shanto) [inline src/MigrationScript.php]: are we only running this script for `v1_0_12`?
what if we release another version and someone updates to that version before updating to `v1_0_12`?

I think this method can be stripped to a single method instead like:

```php
private function runMigration( ) {
    $this->migratePostSyncOption();
}
```

File: readme.txt [hunk 4]
```diff
@@ -223,6 +223,9 @@ Privacy is our utmost priority, and we designed ThriveDesk in a way that aligned
 - Easy setup: Setup your Shared Inbox in less than a minute.
 
 == Changelog ==
+= 1.0.12 =
+- Fix: post sync was not working for rest route
```
Comment (Rijoanul-Shanto) [inline readme.txt:227]: change the changelog 😃

File: src/MigrationScript.php (hunk 1 above)
Comment (atiq-ur) [reply src/MigrationScript.php]: Yeah, we can.

File: database/Scripts/MigrationScript.php [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+	public function run() {
+		$this->runMigrationScript();
+	}
+
+	private function runMigrationScript(  ) {
+		$this->migratePostSyncOption();
+	}
+
+	private function migratePostSyncOption( ) {
+		$thrivedesk_post_type_sync_option = get_option( 'thrivedesk_post_type_sync_option');
```
Comment (Rijoanul-Shanto) [inline database/Scripts/MigrationScript.php:30]: As discussed:

```php
$old_post_sync_option = get_option( 'thrivedesk_post_type_sync_option' );

if ($old_post_sync_option) {
	update_option( 'td_helpdesk_post_sync', $old_post_sync_option );
}

delete_option( 'thrivedesk_post_type_sync_option' );
```

File: readme.txt (hunk 4 above)
Comment (Rijoanul-Shanto) [reply readme.txt:227]: - Fix: Legacy post types support for WPPostSync integration

File: src/Plugins/WPPostSync.php [hunk 6]
```diff
@@ -124,7 +124,7 @@ public function get_post_search_result(string $query_string = ''): array {
 		$x_query      = new \WP_Query(
 			[
 				's'         => $query_string,
-				'post_type' => get_option('thrivedesk_post_type_sync_option'),
+				'post_type' => get_option('td_helpdesk_post_sync'),
```
Comment (Rijoanul-Shanto) [inline src/Plugins/WPPostSync.php:127]: Check for nested relations

File: database/Scripts/MigrationScript.php [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+	}
+
+	private function runMigrationScript(  ) {
+		$this->migratePostSyncOption();
+	}
+
+	private function migratePostSyncOption( ) {
+		$old_post_sync_option = get_option( 'thrivedesk_post_type_sync_option' );
+
+		if ( !$old_post_sync_option ) {
```
Comment (Rijoanul-Shanto) [inline database/Scripts/MigrationScript.php:32]: `if ( !$old_post_sync_option == false )`

**Pull request merged**